### PR TITLE
[docs] document log trace correlation

### PR DESCRIPTION
**Pull request by lachmatt**
Document different ways logs can be correlated with traces.

### CODE REVIEW

**Review by pellared — COMMENTED** [review]
This PR should be "moved" to https://github.com/splunk/public-o11y-docs

CC @theletterf

File: docs/trace-context.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+                rollOnFileSizeLimit: true)
+            .CreateLogger();
+```
+
+Your logs will now have trace context injected into them:
+
+```text
+[16:42:47 INF] Logged inside activity
+|TraceId=de368caa17c4b63154cb8eae99f0db84|SpanId=d2343289a8628728|TraceFlags=Recorded
+```
```
Comment (pellared) [inline docs/trace-context.md:141]: I am not sure if it is needed.

File: docs/trace-context.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+```xml
+<target xsi:type="File" name="logfile" fileName="c:\temp\console-example.log"
+                layout="${longdate}|${level}|${message}|TraceId=${activity:property=TraceId}|SpanId=${activity:property=SpanId}|ParentId=${activity:property=ParentId}|${all-event-properties} ${exception:format=tostring}" />
+```
+
+Your logs will now have trace context injected into them:
+
+```text
+2023-09-14 16:53:25.9139|Info|Logged inside activity|TraceId=23276df3a4a54414d196b88d71338806|SpanId=6e20050fc23d9a2a|TraceFlags=Recorded|  
+```
```
Comment (pellared) [inline docs/trace-context.md:172]:
```suggestion
You can simply use [`NLog.DiagnosticSource`](https://www.nuget.org/packages/NLog.DiagnosticSource).
([documentation](https://github.com/NLog/NLog.DiagnosticSource)).
```

**Comment by Kielek** [comment]
LGTM, I agree that it should go to splunk docs.

**Pull request closed without merging**